### PR TITLE
Add --upgrade-pip-to for increased build stability (closes #266)

### LINKED ISSUE #266: Qustion: using getpip or similar to get a specific pip version?
Is there any way to use a service like getpip to install the preferred pip version?

Background:
we use the extra pip flag `--no-cache-dir` on a debian Jessie machine. The pip version that get's installed doesn't support this flag. To get this to work we've been using `--upgrade-pip` to get us to a python which supports this option.

This is sort of bad since we're always going to end up on the latest pip at some stage during the setup. Today that bit us since the `--no-cache-dir` option broke when pip 19.0.0 was released. And using `--preinstall` to install an older version of pip wouldn't work either since the preinstall command also gets passed the `--no-cache-dir` option.

The --no-cache-dir issue has been resolved upstream from pip==19.0.1 but we're running python 3.4 on our target and pip will not support python 3.4 beyond 19.1 so we'll need to freeze our pip version before then.

The only way I can think of from the top of my head is to create a shell script wrapping the pip call stripping failing arguments if the version is too low...

### PR DESCRIPTION
**Pull request by jhermann**
Includes some minor improvements that I was too lazy to PR separately.

### CODE REVIEW

File: doc/tutorial.rst [hunk 1]
```diff
@@ -229,5 +229,5 @@ Step 3: Build your project
 ==========================
 
 Now you can just build your project by running
-``dpkg-buildpackage -us -uc -b``.
+``( deactivate ; dpkg-buildpackage -us -uc -b )``.
```
Comment (nailor) [inline doc/tutorial.rst]: I'm slightly cautious here: we don't run any virtualenvs prior in this documentation. Now it is entirely possible, that people have an virtualenv activated, which would screw up the build, but not sure if we should default to deactivation? Not entirely sure on my take on this. Maybe separate note to tell people to deactivate if they are already in dh-virtualenv?

File: doc/usage.rst [hunk 2]
```diff
@@ -139,7 +139,17 @@ few command line options:
    because the default ``pip`` of your system might not support them
    (since version 1.1).
 
-   *Note:* This can produce non-repeatable builds.
+   *Note:* This can produce non-repeatable builds. See also :option:`--upgrade-pip-to`.
+
+.. option:: --upgrade-pip-to <VERSION>
```
Comment (nailor) [inline doc/usage.rst]: Should we mark `--upgrade-pip` deprecated and nudge people towards to this, what do you think?

Comment (jhermann) [reply doc/usage.rst]: Only in 1.4+, let people try the new one first.

File: doc/tutorial.rst (hunk 1 above)
Comment (jhermann) [reply doc/tutorial.rst]: this is a secure command variant always succeeding for all people. what's called a winner… ;)

and the effects of not having it when you need it are mostly silent, nasty, and not easily debugged.

File: doc/usage.rst (hunk 2 above)
Comment (nailor) [reply doc/usage.rst]: Good point  👍

File: doc/tutorial.rst (hunk 1 above)
Comment (nailor) [reply doc/tutorial.rst]: That's true!

**Comment by forslund** [comment]
Thanks! will upgrade and test!

**Comment by jhermann** [comment]
BTW, if you add a milestone for 1.4 and add a deprecation issue to it, that won't get lost…

**Pull request merged**